deltacat/autoscaler/events/dispatcher.py

```diff
@@ -0,0 +1,141 @@
+import logging
+from typing import Dict, Any, Optional, Union, Callable
+
+from ray.autoscaler._private.aws.events import AwsEventManagerBase
+from ray.autoscaler._private.event_system import RayEvent, ScriptStartedEvent, ScriptInProgressEvent, \
+    ScriptInProgressCustomEvent, ScriptCompletedEvent, event_enum_values, States
+
+from deltacat import logs
+
+stats_metadata_collection_started_event = ScriptInProgressCustomEvent("STATS_METADATA_COLLECTION_STARTED", 2)
+stats_metadata_collection_completed_event = ScriptInProgressCustomEvent("STATS_METADATA_COLLECTION_COMPLETED", 3)
+compaction_session_started_event = ScriptInProgressCustomEvent("COMPACTION_SESSION_STARTED", 4)
+compaction_session_completed_event = ScriptInProgressCustomEvent("COMPACTION_SESSION_COMPLETED", 5)
+custom_events = [stats_metadata_collection_started_event,
+                 stats_metadata_collection_completed_event,
+                 compaction_session_started_event,
+                 compaction_session_completed_event]
+
+
+logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
+
+
+class EventDispatcher:
+    def __init__(self, events_manager: AwsEventManagerBase):
+        """Constructor for the event dispatcher.
+
+        Intended for usage by Ray parent and child clusters running BDT compaction workflows.
+        This helper class is a wrapper around the events_manager for dispatching compaction workflow events
+        annotated with event metadata for tracking purposes, such as parent and child Ray session IDs.
+
+        Args:
+            events_manager: Events manager for publishing events through a cloud provider
+        """
+        self.events_manager = events_manager
+
+    def dispatch_event(self,
+                       event: RayEvent,
+                       parent_session_id: Optional[str] = None,
+                       session_id: Optional[str] = None,
+                       event_data: Dict[str, Any] = None):
+        if event_data is None:
+            event_data = {}
+
+        event_data["event"] = event
+        event_data["rayParentSessionId"] = parent_session_id
+        event_data["raySessionId"] = session_id
+
+        self._publish_event({
+            **self.events_manager.config["parameters"],
+            **event_data
+        })
+
+    def start_initializing(self):
+        """Publish a job state event
+        """
```

Comment:
Can these docstrings include a description of the intent of each state being published (e.g. the initializing state indicates that a request to start a job run has been successfully received, but the job run has not yet finished prerequisite initialization steps required to make progress on the requested job).